go/pkg/client/batch_retries_test.go

```diff
@@ -289,24 +293,32 @@ func (s *sleepyBatchServer) GetTree(req *repb.GetTreeRequest, stream regrpc.Cont
 }
 
 func (s *sleepyBatchServer) BatchReadBlobs(ctx context.Context, req *repb.BatchReadBlobsRequest) (*repb.BatchReadBlobsResponse, error) {
+	defer s.wg.Done()
+	s.mu.Lock()
 	s.readRequests++
 	s.numErrors++
 	if s.numErrors < 4 {
+		s.mu.Unlock()
```

Comment:
Why not defer the Unlock()?

Reply:
I want the sleep to be outside of the critical section.